HandEyeCalibration/Transform/Transform.cpp

```diff
@@ -0,0 +1,330 @@
+    std::cout << "\nAxis-angle:\n" << reorderAxisAngle(axisAngle) << std::endl;
+
+    // Axis-Angle to Quaternion
+    std::cout << "\nConverting axis-angle to quaternion" << std::endl;
+    const Eigen::Quaterniond quaternion(axisAngle);
+    std::cout << "\nQuaternion:\n" << quaternion.coeffs() << std::endl;
+
+    // Quaternion to Rotation Matrix
+    std::cout << "\nConverting quaternion to rotation matrix" << std::endl;
+    const Eigen::Matrix3d rotationMatrixFromQuaternion = quaternion.toRotationMatrix();
+    std::cout << "\nRotation Matrix:\n" << rotationMatrixFromQuaternion << std::endl;
+    fileStorageOut.write("RotationMatrixFromQuaternion", eigenToCv(rotationMatrixFromQuaternion));
+
+    // Calculate error in transformation (Rotation Matrix --> Quaternion --> Rotation Matrix)
+    std::cout << "\nVerifying Rotation Matrix from Quaternion" << std::endl;
+    compareRotation(rotationMatrix, rotationMatrixFromQuaternion);
+
+    fileStorageOut.release();
+}
+
+std::string toString(RotationConvention convention)
+{
+    switch(convention)
+    {
+        case RotationConvention::XYZ_Intrinsic: return "XYZ_Intrinsic";
+        case RotationConvention::XYZ_Extrinsic: return "XYZ_Extrinsic";
+        case RotationConvention::ZYX_Intrinsic: return "ZYX_Intrinsic";
+        case RotationConvention::ZYX_Extrinsic: return "ZYX_Extrinsic";
+        default: throw std::invalid_argument("Invalid RotationConvention");
+    }
+}
+
+Eigen::Vector4d reorderAxisAngle(const Eigen::AngleAxisd &axisAngle)
+{
+    Eigen::Vector4d reorderedAxisAngle;
+
+    for(int i = 0; i < 3; i++)
+        reorderedAxisAngle[i] = axisAngle.axis()[i];
+    reorderedAxisAngle[3] = axisAngle.angle();
+
+    return reorderedAxisAngle;
+}
+
+/** Converts Rotation Matrix to Roll-Pitch-Yaw angles in radians.
+    * The convention we use here is that Roll is a rotation about x-axis,
+    * Pitch is a rotation about y-axis and Yaw is a rotation about z-axis.
+    * @param rotationMatrix Rotation Matrix to convert to RPY
+    * @param rotation Enum to indicate which rotation convention to follow for this conversion
+    * @return An array of angles, in radians, ordered by Ox, Oy and Oz, where Oa is the rotation about axis a.
+    */
+Eigen::Array3d rotmat2rpy(const Eigen::Matrix3d &rotationMatrix,
+                          const RotationConvention &rotation = RotationConvention::ZYX_Intrinsic)
+{
+    switch(rotation)
+    {
+        case RotationConvention::XYZ_Intrinsic: return rotationMatrix.eulerAngles(0, 1, 2);
+        case RotationConvention::XYZ_Extrinsic: return rotationMatrix.eulerAngles(2, 1, 0).reverse();
+        case RotationConvention::ZYX_Intrinsic: return rotationMatrix.eulerAngles(2, 1, 0).reverse();
+        case RotationConvention::ZYX_Extrinsic: return rotationMatrix.eulerAngles(0, 1, 2);
+        default: throw std::invalid_argument("Invalid rotation");
+    }
+}
+
+/** Converts Roll-Pitch-Yaw angles in radians to Rotation Matrix.
```

Comment:
This comment can be removed, as it contains no extra information compared to the function signature.

Reply:
Also, I don't think you need to note the type of the angles when  using radians. If you for some reason used degrees, it would be a good idea to note it in the variable or function name.

Reply:
That assumes that radians is always the default. I found that RoboDK uses degrees by default, got hurt by it, and decided to explicitly state radians :)

Reply:
When we lack proper types to distinguish radians/degrees, I think it's good to be on the safe side and document that it's in radians.

Reply:
Radians should always be default. When converting for use in some library that uses degrees, you can explicitly state it.
Otherwise, we would have to write `myAngleRad` everywhere.